src/Language/Haskell/Liquid/Constraint/Generate.hs

```diff
@@ -628,51 +622,49 @@ consCBLet :: CGEnv -> CoreBind -> CG CGEnv
 --------------------------------------------------------------------------------
 consCBLet γ cb
   = do oldtcheck <- tcheck <$> get
-       strict    <- specLazy <$> get
-       let tflag  = oldtcheck
-       let isStr  = tcond cb strict
+       lazyVars  <- specLazy <$> get
+       let isStr  = doTermCheck lazyVars cb
        -- TODO: yuck.
-       modify $ \s -> s { tcheck = tflag && isStr }
-       γ' <- consCB (tflag && isStr) isStr γ cb
+       modify $ \s -> s { tcheck = oldtcheck && isStr }
+       γ' <- consCB (oldtcheck && isStr) isStr γ cb
        modify $ \s -> s{tcheck = oldtcheck}
        return γ'
 
 --------------------------------------------------------------------------------
 -- | Constraint Generation: Corebind -------------------------------------------
 --------------------------------------------------------------------------------
-consCBTop :: (Var -> Bool) -> CGEnv -> CoreBind -> CG CGEnv
+consCBTop :: Config -> GhcInfo -> CGEnv -> CoreBind -> CG CGEnv
 --------------------------------------------------------------------------------
-consCBTop trustBinding γ cb | all trustBinding xs
+consCBTop cfg info γ cb
+  | all (trustVar cfg info) xs
   = do ts <- mapM trueTy (varType <$> xs)
        foldM (\γ xt -> (γ, "derived") += xt) γ (zip xs' ts)
     where
        xs  = bindersOf cb
        xs' = F.symbol <$> xs
 
-consCBTop _ γ cb
+consCBTop _ _ γ cb
   = do oldtcheck <- tcheck <$> get
-       strict    <- specLazy <$> get
-       let tflag  = oldtcheck
-       let isStr  = tcond cb strict
-       modify $ \s -> s { tcheck = tflag && isStr}
-
+       lazyVars  <- specLazy <$> get
+       let isStr  = doTermCheck lazyVars cb
+       modify $ \s -> s { tcheck = oldtcheck && isStr}
        -- remove invariants that came from the cb definition
-       let (γ',i) = removeInvariant γ cb
-       γ'' <- consCB (tflag && isStr) isStr γ' cb
+       let (γ', i) = removeInvariant γ cb                 --- DIFF
+       γ'' <- consCB (oldtcheck && isStr) isStr γ' cb
        modify $ \s -> s { tcheck = oldtcheck}
-       return $ restoreInvariant γ'' i
+       return $ restoreInvariant γ'' i                    --- DIFF
 
+-- HEREHEREHEREHERE
+-- why is consCBTop ALMOST equivalent to consCBLet ? except for this removeInvariant stuff?
```

Comment:
remove invariant removes the measure invariant (post-condition) when the measure is checked